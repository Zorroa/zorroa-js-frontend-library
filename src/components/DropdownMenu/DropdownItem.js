import React, { PropTypes, PureComponent } from 'react'
import classnames from 'classnames'

class DropdownItem extends PureComponent {
  static propTypes = {
    dark: PropTypes.bool.isRequired,
    onClick: PropTypes.func,
    children: React.PropTypes.any,
  }

  hasHandler() {
    return typeof this.props.onClick === 'function'
  }

  handleClick = event => {
    if (this.hasHandler()) {
      this.props.onClick(event)
    }
  }

  isDark() {
    return this.props.dark === true
  }

  render() {
    const dropdownItemCLasses = classnames('DropdownMenu__item', {
      'DropdownMenu__item--dark': this.isDark(),
      'DropdownMenu__item--disabled': !this.hasHandler(),
    })
    return (
      <div className={dropdownItemCLasses} onClick={this.handleClick}>
        {this.props.children}
      </div>
    )
  }
}

export default DropdownItem
