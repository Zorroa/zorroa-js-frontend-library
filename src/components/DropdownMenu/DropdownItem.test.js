/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import DropdownItem from './DropdownItem'

describe('<DropdownItem />', () => {
  describe('handleClick()', () => {
    it('Should not crash when no click handler is defined', () => {
      const event = {}
      const component = shallow(<DropdownItem dark={true} />)
      component.instance().handleClick(event)
    })

    it('Should call the click handler', () => {
      const onClick = jest.fn()
      const event = {}
      const component = shallow(<DropdownItem onClick={onClick} dark={true} />)
      component.instance().handleClick(event)
      expect(onClick.mock.calls.length).toBe(1)
    })
  })

  describe('render()', () => {
    describe('When there is no handler', () => {
      it('Should have a disabled class', () => {
        const tree = renderer
          .create(<DropdownItem dark={false}>Move Left</DropdownItem>)
          .toJSON()
        expect(tree).toMatchSnapshot()
      })
    })

    describe('When in dark mode', () => {
      it('Should have a dark class', () => {
        const tree = renderer
          .create(<DropdownItem dark={true}>Preferences</DropdownItem>)
          .toJSON()
        expect(tree).toMatchSnapshot()
      })
    })
  })
})
