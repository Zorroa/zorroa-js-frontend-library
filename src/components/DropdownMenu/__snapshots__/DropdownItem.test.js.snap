// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem /> render() When in dark mode Should have a dark class 1`] = `
<div
  className="DropdownMenu__item DropdownMenu__item--dark DropdownMenu__item--disabled"
  onClick={[Function]}
>
  Preferences
</div>
`;

exports[`<DropdownItem /> render() When there is no handler Should have a disabled class 1`] = `
<div
  className="DropdownMenu__item DropdownMenu__item--disabled"
  onClick={[Function]}
>
  Move Left
</div>
`;
